api/envoy/config/core/v3/protocol.proto

```diff
@@ -123,6 +123,24 @@ message UpstreamHttpProtocolOptions {
       [(validate.rules).string = {well_known_regex: HTTP_HEADER_NAME ignore_empty: true}];
 }
 
+// Allows pre-populating the cache with HTTP/3 alternate protocols entries with a 7 day lifetime.
+// This will cause Envoy to attempt HTTP/3 to those upstreams, even if the upstreams have not
+// advertised HTTP/3 support. These entries will be overwritten by alt-svc
+// response headers or cached values.
+// As with regular cached entries, if the origin response would result in clearing an existing
+// alternate protocol cache entry, pre-populated entries will also be cleared.
+// Adding a cache entry with hostname=foo.com port=123 is the equivalent of getting
+// response headers
+// alt-svc: h3=:"123"; ma=86400" in a response to a request to foo.com:123
+message AlternateProtocolsCacheEntry {
```

Comment:
Should this be nested inside of `AlternateProtocolsCacheOptions`?  Or is there a reason to think this might be used somewhere else in the future?

My concern here is that we seem to have a bunch of messages that are HTTP-specific but do not have "http" in the name anywhere, which seems like a bit of namespace pollution.  I would have chosen a different name for `AlternateProtocolsCacheOptions` too, but at least that message already exists here, so nesting this new message inside of it avoids making the problem worse.

Reply:
I can move it over if we prefer nesting.  This is unlikely to be used elsewhere but I've seen nesting be problematic often enough I generally lean against.  Your call!

Reply:
I'd prefer to nest it if we don't think it's going to be used elsewhere.  Thanks!